### DISCUSSION TITLE
How do I make sure an error from a resource gets caught by a higher level error boundary?

### DISCUSSION THREAD

**Discussion by SleeplessOne1917** [leptos-rs/leptos#1862] [Q&A] [answered]
Suppose I have a component like this:
```rust
#[component]
fn MyComponent() -> impl IntoView {
    let my_resource = create_resource(/* resource fetching stuff */); // resource for a result

    provide_context(my_resource);

    view! {
        <Suspense>
            <ErrorBoundary> {/* I want to catch errors for the resource here*/}
                {/* Several nested components later*/}
                    <ErrorBoundary>
                        <NestedComponent /> {/* This component uses the resource defined above via use_context*/}
                    </ErrorBoundary>
                {/* Several nested components later*/}
            </ErrorBoundary>
        </Suspense>
    }
}
```

How do I make sure an error from the resource gets handled at the error boundary I marked in the code snippet?

**Comment by gbj** [comment, marked as answer]
Very simply: an `ErrorBoundary` is triggered if you render an `Err(_)` inside it.

So:
1. To trigger the parent boundary: inside that desired boundary, do something like `my_resource.and_then(|_| ())` to render either `Ok(())` (shows nothing) or `Err(_)` (triggers error boundary)
2. To avoid triggering the child boundaries: `.unwrap_or_else()` the Result to some default value such that it can never render `Err` under those children
